plugins/kubernetes/src/components/K8sTerminal/K8sTerminal.test.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import './matchMedia.mock';
+import '@testing-library/jest-dom';
+
+import { DiscoveryApi, discoveryApiRef } from '@backstage/core-plugin-api';
+import {
+  renderWithEffects,
+  TestApiProvider,
+  textContentMatcher,
+  wrapInTestApp,
+} from '@backstage/test-utils';
+import WS from 'jest-websocket-mock';
+import React from 'react';
+
+import { K8sTerminal } from './K8sTerminal';
+
+const textEncoder = new TextEncoder();
+
+describe('K8sTerminal', () => {
+  const clusterName = 'cluster1';
+  const containerName = 'container2';
+  const podName = 'pod1';
+  const podNamespace = 'podNamespace';
+
+  const mockDiscoveryApi: Partial<DiscoveryApi> = {
+    getBaseUrl: () => Promise.resolve('http://localhost'),
+  };
+
+  it('Should render an XTerm web shell', async () => {
+    const { getByText } = await renderWithEffects(
+      wrapInTestApp(
+        <TestApiProvider apis={[[discoveryApiRef, mockDiscoveryApi]]}>
+          <K8sTerminal
+            clusterName={clusterName}
+            containerName={containerName}
+            podName={podName}
+            podNamespace={podNamespace}
+          />
+        </TestApiProvider>,
+      ),
+    );
+
+    expect(
+      getByText(textContentMatcher('Starting terminal, please wait...')),
+    ).toBeInTheDocument();
+  });
+
+  it('Should connect to WebSocket server & render response', async () => {
+    const server = new WS(
+      'ws://localhost/proxy/api/v1/namespaces/podNamespace/pods/pod1/exec?container=container2&stdin=true&stdout=true&stderr=true&tty=true&command=%2Fbin%2Fsh',
+    );
+
+    const { getByText } = await renderWithEffects(
+      wrapInTestApp(
+        <TestApiProvider apis={[[discoveryApiRef, mockDiscoveryApi]]}>
+          <K8sTerminal
+            clusterName={clusterName}
+            containerName={containerName}
+            podName={podName}
+            podNamespace={podNamespace}
+          />
+        </TestApiProvider>,
+      ),
+    );
+
+    // Measure element, if it exists the XTerm shell rendered correctly
+    expect(getByText(textContentMatcher('W'))).toBeInTheDocument();
+
+    await server.connected;
+
+    const buffer = Uint8Array.from([1, ...textEncoder.encode('hello world')]);
+
+    server.send(buffer);
+
+    expect(getByText(textContentMatcher('hello world'))).toBeInTheDocument();
```

Comment:
have you tried with 

```ts
 expect(await findByText(textContentMatcher('hello world'))).toBeInTheDocument()
```

or

```ts
await waitFor(() => expect(getByText(textContentMatcher('hello world'))).toBeInTheDocument())
```
?

it might help with the issue since the rendering is asynchronous

Reply:
Btw I did test for that case and used a set timeout to run `debug()` to log the HTML content and it had not changed with a 5s timeout, so I believe is not an async issue.

Reply:
Never mind, @jamieklassen found a solution with the library.